src/crimson/os/seastore/async_cleaner.h

```diff
@@ -963,38 +965,72 @@ class AsyncCleaner : public SegmentProvider, public Trimmer {
     assert(init_complete);
     auto ret = journal_head;
     ceph_assert(ret != JOURNAL_SEQ_NULL);
-    if (ret.segment_seq >= config.target_journal_dirty_segments) {
-      ret.segment_seq -= config.target_journal_dirty_segments;
+    if (get_journal_type() ==
+      journal_type_t::SEGMENT_JOURNAL) {
+      if (ret.segment_seq >= config.target_journal_dirty_segments) {
+	ret.segment_seq -= config.target_journal_dirty_segments;
+	return ret;
+      }
     } else {
-      ret.segment_seq = 0;
-      ret.offset = P_ADDR_MIN;
+      ceph_assert(get_journal_type() ==
+	journal_type_t::CIRCULARBOUNDED_JOURNAL);
+      auto gap = get_len_to_head(get_dirty_tail());
+      if (gap > config.rewrite_dirty_bytes_per_cycle) {
```

Comment:
I don't understand the usage of `rewrite_dirty_bytes_per_cycle` here, seems the config is irrelavent here, same in `get_tail_limit()` and `get_alloc_tail_target()`.